server/src/config/connectionStringBuilder.js

```diff
@@ -0,0 +1,34 @@
+module.exports = class MongoConnection {
+  constructor() {
+    this.conn = this.getConnection();
+    this.uri = this.buildUri();
+    this.databaseName = this.conn.databaseName;
+    this.reconnectInterval = 10000;
+  }
+
+    getConnection = () => {
+      const conn = {
+        username: process.env.MONGO_USERNAME,
+        passwd: process.env.MONGO_PASSOWORD,
+        host: process.env.MONGO_HOST,
+        port: process.env.MONGO_PORT,
+        databaseName: process.env.MONGO_DATABASE,
+      };
+
+      const undefinedFields = Object.keys(conn).filter((item) => !conn[item]);
+      if (undefinedFields.length > 0) {
+        throw new Error(`Some of mandatory environmental variables are missing [${undefinedFields}]`);
+      }
+
+      return conn;
+    }
+
+    buildUri = () => {
+      const uri = 'mongodb://'
+            + `${this.conn.username}:${encodeURIComponent(this.conn.passwd)}@`
+            + `${this.conn.host}:${this.conn.port}`
+            + '/?maxPoolSize=20&w=majority';
```

Comment:
I think it would make sense to have this part as env variable as well, with a default value as above - just to have the option to tinker with it without rebuilding the app

Reply:
Good point, moved to separate variable.